Plugins/Skinning/src/PositionBasedSkinning/PBS.cpp

```diff
@@ -0,0 +1,521 @@
+        Scalar distanceBone = distancePointSegment(particle, aStartBone, aEndBone);
+        if (distanceBone < distanceNearestBone)
+        {
+            indexNearestBone = i;
+            distanceNearestBone = distanceBone;
+            startBone = aStartBone;
+            endBone = aEndBone;
+        }
+    }
+}
+
+void PBS::weightParticlesBindConstraint ()
+{
+    unsigned int indexNearestBone;
+    Scalar distanceNearestBone;
+    Vector3 startBone, endBone;
+    ParticleData &pd = m_model.getParticles();
+    const unsigned int num_particles = pd.size();
+
+    m_weightParticles.resize(num_particles, m_skeleton->size());
+    m_weightParticles.setZero();
+
+    for (auto i = 0; i < num_particles; ++i)
+    {
+        Vector3 particle_position = pd.getPosition(i);
+        searchNearestBone(particle_position, startBone, endBone, indexNearestBone, distanceNearestBone);
+
+        // Attribute Weight
+        weightParticle(m_skeleton->m_graph, i, particle_position, indexNearestBone, startBone, endBone);
+
+        // Add Bind Constraint
+        m_model.addBindConstraint(i, indexNearestBone, distanceNearestBone);
+    }
+}
+
+/* ----------------------
+  Add Constraints
+ ----------------------- */
+
+void PBS::addConstraint_stretch ()
+{
+    unsigned int endPoint1, endPoint2;
+    TetModel *tetrahedra = m_model.getTetModel();
+    TetModel::Edges set_edges = tetrahedra->getEdges();
+
+    // Parallelisme ?
+    for (auto it_edges = set_edges.cbegin(); it_edges != set_edges.cend(); ++it_edges)
+    {
+       endPoint1 = it_edges->m_vert[0];
+       endPoint2 = it_edges->m_vert[1];
+       m_model.addDistanceConstraint(endPoint1, endPoint2);
+    }
+}
+
+void PBS::addConstraint_volume ()
+{
+    TetModel *tetrahedra = m_model.getTetModel();
+    TetModel::Tets tets = tetrahedra->getTets();
+
+    for (auto it_tet = tets.cbegin(); it_tet != tets.cend(); it_tet += 4)
+    {
+        m_model.addVolumeConstraint(*it_tet, *(it_tet + 1), *(it_tet + 2), *(it_tet + 3));
+    }
+}
+
+
+//-----------------------------------------------------------------------------------------------------------
+
+void PBS::compute ( const Animation::Skeleton *skeleton, const Animation::Pose &pose,
+                     Ra::Core::Vector3Array &outMesh)
+{
+    if (m_volumetricMeshValid)
+    {
+        ParticleData &pd = m_model.getParticles();
+         linearBlendSkinning(pose, pd);
+         step(skeleton, pd);
+         updateVertices(outMesh);
+
+        /*for (auto i = 0; i < pd.size(); ++i)
+        {
+                 RA_DISPLAY_POINT( pd.getPosition(i), Ra::Core::Colors::Red(), 0.1f );
+        }*/
+    }
+    else
+    {
+
+        QMessageBox messageBox;
+        messageBox.critical(0,"Tetrahedral Mesh"," Tetrahedral mesh must be loaded first!");
+        messageBox.setFixedSize(500,200);
+    }
+
+}
+
+void PBS::linearBlendSkinning(const Animation::Pose &pose, ParticleData &pd)
+{
+    pd.resetCurrentPositions();
+
```

Comment:
can you use Ra::Core::Animation::linearBlendSkinning() ?

Reply:
this LBS bind based on the previous position of the skeleton.